src/natcap/invest/coastal_vulnerability.py

```diff
@@ -2669,78 +2790,50 @@ def _aggregate_raster_values_in_radius(
         if pixel_y + win_ysize > n_rows:
             warn_msg = 1
             win_ysize -= pixel_y + win_ysize - n_rows
             # trim mask rows by indexing forward from top
-            temp_kernel_mask = temp_kernel_mask[:win_ysize, :]
+            kernel_mask = kernel_mask[:win_ysize, :]
         if warn_msg:
             LOGGER.warning(
                 'search radius around point (%.6f, %.6f) '
                 'extends beyond bounds of raster %s',
                 point_x, point_y, base_raster_path)
         if win_xsize <= 0 or win_ysize <= 0:
-            pixel_value = 0
-        else:
-            try:
-                array = band.ReadAsArray(
-                    xoff=pixel_x, yoff=pixel_y, win_xsize=win_xsize,
-                    win_ysize=win_ysize)
-                if nodata is not None:
-                    mask = ~numpy.isclose(array, nodata) & temp_kernel_mask
-                else:
-                    mask = kernel_mask
-                if numpy.count_nonzero(mask) > 0:
-                    pixel_value = numpy.mean(array[mask])
-                else:
-                    pixel_value = numpy.nan
-            except Exception:
-                LOGGER.error(
-                    'band size %d %d', band.XSize,
-                    band.YSize)
-                raise
-
-        # calculate pixel area in sq km
-        if aggregation_mode == 'density' and ~numpy.isnan(pixel_value):
-            pixel_area_km = abs(
-                geotransform[1] * geotransform[5]) / 1e6  # converts m^2 to km^2
-            pixel_value /= pixel_area_km
-
-        result[shore_id] = pixel_value
-
-    n_missing_data = numpy.count_nonzero(numpy.isnan(list(result.values())))
-    if n_missing_data:
-        LOGGER.warning(
-            "\n**************************************************\n"
-            "%d points have a missing value after aggregating %s because no "
-            "valid pixels were found within the search radius (%d meters) "
-            "around these points. You may wish to increase the search radius "
-            "input parameter, or you may wish to view %s and the landmass "
-            "polygon input in a GIS to confirm that both layers are accurate "
-            "and well-aligned. "
-            "\n**************************************************"
-            % (n_missing_data, os.path.basename(base_raster_path),
-                sample_distance, os.path.basename(base_raster_path)))
+            # Not sure how we get here, but clamping to 0 allows
+            # the aggregation_op to decide what to return given
+            # an empty array.
+            win_xsize = 0
+            win_ysize = 0
+        array = band.ReadAsArray(
+            xoff=pixel_x, yoff=pixel_y, win_xsize=win_xsize,
+            win_ysize=win_ysize)
+        if nodata is not None:
+            kernel_mask &= ~numpy.isclose(array, nodata)
+
+        result[shore_id] = aggregation_op(array, kernel_mask)

```

Comment:
I see that in the removed code, there's a check in the `'density'` op about whether there are `nan` pixels.  Is it expected to have `result[shore_id] = numpy.nan`, for the case that one of the ops does end up returning `nan`?

Reply:
Yes, `nan` values are expected to exist in those `result` dicts, if that is your question.

Reply:
There are tests to make sure `nan` is handled properly later on when these intermediate results are used in subsequent equations.